cms/jinja2/templates/components/streamfield/table_block.html

```diff
@@ -1,2 +1,13 @@
-{% from "components/table/_macro.njk" import onsTable %}
-{{ onsTable(options) }}
+{% from "component_overrides/table/_macro.njk" import onsTable %}
+<figure id="table-{{ block_id }}">
+    {% if title %}<h3 class="heading-4">{{ title }}</h3>{% endif %}
+
+    {{ onsTable(options) }}
+
+    {%- if source or footnotes %}
+        <footer>
+            {% if source %}<p>{{ _("Source") }}: {{ source }}</p>{% endif %}
+            {% if footnotes %}{{ footnotes|richtext }}{% endif %}
+        </footer>
+    {% endif %}
+</figure>
```

Comment:
Currently, publishing is possible without any table data. We should either prevent saving the page when there is no table data or only display the supporting element when table data exists like `cms/tiny_table_block/templates/tiny_table_block/table_block.html` does.

Reply:
Funny enough, I have been thinking about the same thing this morning. Will sort

Reply:
5139d29
went with hide if empty table